src/lib/WalletManager.ts

```diff
@@ -0,0 +1,136 @@
+import { deleteItemAsync, getItemAsync, setItemAsync, WHEN_UNLOCKED } from 'expo-secure-store'
+import { isValidDeviceId, StoredWallet, StoredWalletWithMnemonic, Wallet } from './Wallet'
+
+const getKey = (key: string) => {
+  if (!isValidDeviceId(key)) throw new Error('Invalid key')
+  return `mnemonic_${key}`
+}
+
+enum UpdateAction {
+  ADD,
+  REMOVE,
+}
+
+export class WalletManager {
+  #index = new Set<string>()
+
+  public readonly [Symbol.toStringTag]: string = 'WalletManager'
+
+  public async initialize() {
+    try {
+      if (this.#index.size === 0) {
+        const index: string[] = JSON.parse((await getItemAsync('mnemonic-index')) || '[]')
+        this.#index = new Set(index)
+      } else {
+        console.error('[WalletManager.initialize] initialize was already called')
+      }
+    } catch (e) {
+      console.error('[WalletManager.initialize] Unable to read wallet index')
```

Comment:
should we return errors here for direction to the error page?

Reply:
Good question. We could just set the index to an empty set and overwrite the bad one. If we can't find parse the JSON then I don't think there's anything we can do about it expect replace the index with an empty set. 

What do you think?